net/libfab/libfab_internal.h

```diff
@@ -0,0 +1,609 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#pragma once
+
+#ifndef __MOTR_NET_LIBFAB_LIBFAB_INTERNAL_H__
+#define __MOTR_NET_LIBFAB_LIBFAB_INTERNAL_H__
+
+#ifdef ENABLE_LIBFAB
+
+#include <netinet/in.h>                  /* INET_ADDRSTRLEN */
+
+#include "rdma/fabric.h"
+#include "rdma/fi_cm.h"
+#include "rdma/fi_domain.h"
+#include "rdma/fi_eq.h"
+#include "rdma/fi_endpoint.h"
+#include "rdma/fi_rma.h"
+
+extern struct m0_net_xprt m0_net_libfab_xprt;
+
+/**
+ * @defgroup netlibfab
+ *
+ * @{
+ */
+
+#define LIBFAB_VERSION FI_VERSION(1,11)
+
+#define LIBFAB_ADDR_LEN_MAX	INET6_ADDRSTRLEN
+#define LIBFAB_PORT_LEN_MAX	6
+#define LIBFAB_ADDR_STRLEN_MAX  (LIBFAB_ADDR_LEN_MAX + LIBFAB_PORT_LEN_MAX + 1)
+
+/**
+ * Parameters required for libfabric configuration
+ */
+enum m0_fab__libfab_params {
+	/** Fabric memory registration access. */
+	FAB_MR_ACCESS                  = (FI_READ | FI_WRITE | FI_RECV |
+					  FI_SEND | FI_REMOTE_READ |
+					  FI_REMOTE_WRITE),
+	/** Fabric memory registration offset. */
+	FAB_MR_OFFSET                  = 0,
+	/** Fabric memory registration flag. */
+	FAB_MR_FLAG                    = 0,
+	/** Key used for memory registration. */
+	FAB_MR_KEY                     = 0xABCD,
+	/** Max number of IOV in read/write command for Verbs */
+	FAB_VERBS_IOV_MAX              = 1,
+	/** Max segment size for bulk buffers for Verbs */
+	FAB_VERBS_MAX_BULK_SEG_SIZE    = 1048576,
+	/** Max number of active work requests for Verbs */
+	FAB_VERBS_MAX_QUEUE_SIZE       = 224,
+	/** Max number of iov that can be sent in fi_sendmsg() for Verbs */
+	FAB_VERBS_MAX_IOV_PER_TX       = 1,
+
+	/** Max number of IOV in read/write command for TCP/Socket provider
+	 * (max number of segments) */
+	FAB_TCP_SOCK_IOV_MAX           = 256,
+	/** Max segment size for bulk buffers for TCP/Socket provider
+	 * (4k but can be increased) */
+	FAB_TCP_SOCK_MAX_BULK_SEG_SIZE = 4096,
```

Comment:
This to be increased? testing was done with 64K

Reply:
On st 01 testing,
with 4K:
avg out: 16.5MiB/s avg in: 16.4MiB/s total out: 16.5MiB total in: 16.5MiB avg RTT:   1211.693 us 
with 64K
avg out:  653MiB/s avg in:  652MiB/s total out: 1.28GiB total in: 1.28GiB avg RTT:  12120.596 us

Reply:
For S3 IOs, getting ~3.8GBps for reads and ~1.5GBps for writes with seg_size=4k on HW setup which is same as the throughput observed with seg_size=64k. Hence keeping it as 4k.